### PR TITLE
mzidentml

### PR DESCRIPTION
**Pull request by lschaffer2**
It's a start - would like to know if you have any comments/things to add or change.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,233 +1,233 @@
-## Ignore Visual Studio temporary files, build results, and
-## files generated by popular Visual Studio add-ons.
-
```
Comment (stefanks) [inline .gitignore]: What's going on here?

File: TaskLayer/CalibrationTask/CalibrationTask.cs [hunk 2]
```diff
@@ -235,7 +235,7 @@ protected override MyTaskResults RunSpecific(string OutputFolder, List<DbForTask
 
                 InitiatorMethionineBehavior initiatorMethionineBehavior = InitiatorMethionineBehavior.Variable;
                 // Run analysis on single file results
-                var analysisEngine = new AnalysisEngine(searchResults.OuterPsms, compactPeptideToProteinPeptideMatching, proteinList, variableModifications, fixedModifications, localizeableModifications, Protease, searchModes, myMsDataFile, ProductMassTolerance, (BinTreeStructure myTreeStructure, string s) => WriteTree(myTreeStructure, OutputFolder, Path.GetFileNameWithoutExtension(origDataFileName) + "_" + s, new List<string> { taskId, "Individual Searches", origDataFileName }), (List<NewPsmWithFdr> h, string s, List<string> ss) => WritePsmsToTsv(h, OutputFolder, Path.GetFileNameWithoutExtension(origDataFileName) + s, ss), null, false, false, false, MaxMissedCleavages, MinPeptideLength, MaxPeptideLength, MaxModificationIsoforms, false, lp, double.NaN, initiatorMethionineBehavior, new List<string> { taskId, "Individual Searches", origDataFileName }, false, 0, 0);
```
Comment (stefanks) [inline TaskLayer/CalibrationTask/CalibrationTask.cs]: Is the only change here the renaming of WritePsmsToTsv to WritePsms?

File: TaskLayer/EverythingRunner/EverythingRunnerEngine.cs [hunk 3]
```diff
@@ -109,7 +109,7 @@ where currentRawDataFilenameList.All(f => f.StartsWith(possibleMatch, StringComp
 
         #endregion Public Methods
 
-        #region Private Methods
+         #region Private Methods
```
Comment (stefanks) [inline TaskLayer/EverythingRunner/EverythingRunnerEngine.cs]: Not needed!

File: TaskLayer/MetaMorpheusTask.cs [hunk 4]
```diff
@@ -226,9 +237,302 @@ protected internal void WritePsmsToTsv(List<NewPsmWithFdr> items, string outputF
             SucessfullyFinishedWritingFile(writtenFile, nestedIds);
         }
 
-        #endregion Protected Internal Methods
+        protected internal void WritePsmsToMzIdentmL(List<NewPsmWithFdr> items, List<DbForTask> dbFilenameList, string outputFolder, string fileName, List<string> nestedIds)
+        {
+            XmlSerializer _indexedSerializer = new XmlSerializer(typeof(mzIdentML.Generated.MzIdentMLType));
+            var _mzid = new mzIdentML.Generated.MzIdentMLType()
+            {
+                version = "1.2.0"
```
Comment (stefanks) [inline TaskLayer/MetaMorpheusTask.cs]: Are you sure mzlib can write mzidentml version 1.2.0? I think it currently does 1.1.1

File: .gitignore (hunk 1 above)
Comment (lschaffer2) [inline .gitignore]: no idea

File: TaskLayer/CalibrationTask/CalibrationTask.cs (hunk 2 above)
Comment (lschaffer2) [inline TaskLayer/CalibrationTask/CalibrationTask.cs]: Yes that's one method that then calls methods for both the .tsv and .mzidentml

**Pull request closed without merging**